ferrocene/doc/sphinx-shared-resources/exts/ferrocene_domain_cli/domain.py

```diff
@@ -105,6 +131,28 @@ def id(self):
 
         return f"um_{self.program}_{option}"
 
+class Subcommand:
+    has_content = True
+    required_arguments = 1
+    final_argument_whitespace = True
+
+    def __init__(self, document, program_name, subcommand, no_traceability_matrix):
+        self.document = document
+        self.program = program_name
+        self.subcommand = subcommand
+        self.no_traceability_matrix = no_traceability_matrix
+
+    def id(self):
+        subcommand = (
+            ARGUMENT_PLACEHOLDER_RE.sub("_", self.subcommand)
+            .replace("=", "")
+            .replace("-", "_")
+            .replace(" ", "_")
+            .strip("_")
+        )
+        subcommand = MULTIPLE_UNDERSCORES_RE.sub("_", subcommand)
+
+        return f"um_{self.program}_{subcommand}"
```

Comment:
So I am a bit curious how this will work (if at all) when we document programs with subcommands with options.

Looking at the "Option" class, will `criticalup install --project .` and `criticalup verify --project .` have the same `um_criticalup_project` id? Will this be a problem? 🤔

Reply:
Oooh This is a great observation. I think for subcommand I can update the ID to 

```
**f"um_{self.program}_subcommand_{subcommand}"**
```

But I don't know how it will interact with the rest of the codebase. :grimacing: But let me try.